src/core/agent/services/connectionService.test.ts

```diff
@@ -265,6 +253,62 @@ describe("Connection service of agent", () => {
     expect(connectionStorage.save).toBeCalled();
   });
 
+  test("Should throw an error if invalid OOBI URL format", async () => {
+    Agent.agent.getKeriaOnlineStatus = jest.fn().mockReturnValue(true);
+    let invalidUrls = [
+      "https://localhost/oobi",
+      "https://localhost/oobi/1234",
+      "https://localhost/oobi/1234/agent/eid/extra",
+      "https://localhost/.well-known/keri/oobi/",
+    ];
+
+    for (const url of invalidUrls) {
+      await expect(
+        connectionService.connectByOobiUrl(url)
+      ).rejects.toThrowError(new Error(ConnectionService.OOBI_INVALID));
+    }
+
+    invalidUrls = ["https://localhost/oobi"];
+
+    for (const url of invalidUrls) {
+      await expect(connectionService.resolveOobi(url)).rejects.toThrowError(
+        new Error(ConnectionService.OOBI_INVALID)
+      );
+    }
+  });
+
+  test("Should create connection and resolveOOBI with valid URL format", async () => {
+    Agent.agent.getKeriaOnlineStatus = jest.fn().mockReturnValue(true);
+    const validUrls = [
+      "https://example.com/oobi/1234/agent?name=alias",
+      "https://example.com/oobi/1234/agent/5678?name=alias",
+      "https://example.com/.well-known/keri/oobi/1234?name=alias",
+    ];
+
+    for (const url of validUrls) {
+      await connectionService.connectByOobiUrl(url);
+      expect(connectionStorage.save).toBeCalled();
+    }
+
+    signifyClient.operations().get = jest
+      .fn()
+      .mockResolvedValue({ done: true });
+    for (const url of validUrls) {
```

Comment:
We should also test the other valid URLs here for `resolveOobi`:
- schema URL
- OOBI URL where the role is witness and not agent